assets/charts/control-plane/kube-apiserver/templates/kube-apiserver.yaml

```diff
@@ -25,6 +25,8 @@ spec:
       annotations:
         checkpointer.alpha.coreos.com/checkpoint: "true"
         seccomp.security.alpha.kubernetes.io/pod: 'docker/default'
+        # Automatically rolls update when secret changes.
+        checksum/secret: {{ include (print $.Template.BasePath "/kube-apiserver-secret.yaml") . | sha256sum }}
```

Comment:
Not sure why the other components other than the kube-controller-manager have this. Don't we want to:
* Update secrets with new CA
* Restart kube-controller-manager, so it picks up the new CA
* Restart the rest of the control plane pods, that for some reason need to be re-started after kube-controller-manager
* End

Why restart all the rest in the second step too? Does it help in some way to minimize downtime or something?

Reply:
I'm not sure I understand. We only restart kube-controller-manager and kube-apiserver. API server do not sure service account token IIRC.

We do this for `kube-controller-manager` to rollout steps:
- Update secrets with new CA
- Restart kube-controller-manager, so it picks up the new CA

After this is done, `kube-controller-manager` will generate new secrets and at this point we should restart everything using service accounts, including `kube-controller-manager` itself.

`kube-apiserver` should be restarted after certificates change, as it's also handy in case you need to add IP address to the certificate. The CA certificate does not change then, so whole rotation process is not needed then.

Reply:
Ok, not sure I follow, do you say we need to restart kube-apiserver in step 2 (of my previous comment) in case IP address are added to certificates? The simpler steps I proposed won't work in that case too?

Reply:
I'll leave this for now. We can optimize later.